bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/DPIZoomChangeRegistry.java

```diff
@@ -44,6 +44,9 @@ public class DPIZoomChangeRegistry {
 	 * @param scalingFactor factor as division between new zoom and old zoom, e.g. 1.5 for a scaling from 100% to 150%
 	 */
 	public static void applyChange(Widget widget, int newZoom, float scalingFactor) {
+		if (widget == null) {
+			return;
+		}
```

Comment:
Is this really needed? I mean is some call passing `null` as a parameter?

Reply:
Yes, decorations.menus contains null values in some scenarios. And it is definitely a source for possible NPE in the future, so I think fetching nulls here is better